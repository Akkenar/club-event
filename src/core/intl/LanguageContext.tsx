import * as React from 'react';
import { IntlType } from './intl.type';

const LanguageContext = React.createContext<IntlType>({
  confirmation: '',
  information: '',
  messages: {},
});

export default LanguageContext;
