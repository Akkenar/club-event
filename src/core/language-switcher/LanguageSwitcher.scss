@import url('../../../node_modules/semantic-ui-css/components/dropdown.min.css');

.LanguageSwitcher .text {
  width: 2em;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
