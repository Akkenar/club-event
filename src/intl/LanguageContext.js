import React from 'react';
import { DEFAULT_LANGUAGE } from './provideLanguageInRoute';

const LanguageContext = React.createContext({
  messages: {},
  language: DEFAULT_LANGUAGE,
});

export default LanguageContext;
