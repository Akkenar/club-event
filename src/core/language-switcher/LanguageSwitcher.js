import React, { useContext } from 'react';
import getKey from '../intl/getKey';
import { Dropdown } from 'semantic-ui-react';
import LanguageContext from '../intl/LanguageContext';
import { NavLink } from 'react-router-dom';

import './LanguageSwitcher.scss';

const LanguageSwitcher = () => {
  const { handleChangeLanguage, messages } = useContext(LanguageContext);

  return (
    <Dropdown
      className="link item right LanguageSwitcher"
      text={getKey('current.language', messages)}
    >
      <Dropdown.Menu>
        <Dropdown.Item
          title="FranÃ§ais"
          as={NavLink}
          to="/fr"
          onClick={() => handleChangeLanguage('fr')}
        >
          FR
        </Dropdown.Item>
        <Dropdown.Item
          title="Deutsch"
          as={NavLink}
          to="/de"
          onClick={() => handleChangeLanguage('de')}
        >
          DE
        </Dropdown.Item>
        <Dropdown.Item
          title="Italianno"
          as={NavLink}
          to="/it"
          onClick={() => handleChangeLanguage('it')}
        >
          IT
        </Dropdown.Item>
        <Dropdown.Item
          title="English"
          as={NavLink}
          to="/en"
          onClick={() => handleChangeLanguage('en')}
        >
          EN
        </Dropdown.Item>
      </Dropdown.Menu>
    </Dropdown>
  );
};

export default LanguageSwitcher;
