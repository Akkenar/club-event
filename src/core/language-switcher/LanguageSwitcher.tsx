import * as React from 'react';
import { useContext } from 'react';
import { Dropdown } from 'semantic-ui-react';
import getKey from '../intl/getKey';
import LanguageContext from '../intl/LanguageContext';

import './LanguageSwitcher.scss';

const Button = (props: any) => (
  <button
    {...props}
    className={`${props.className || ''} LanguageSwitcher__button`}
  />
);

const LanguageSwitcher = () => {
  const { handleChangeLanguage, messages } = useContext(LanguageContext);

  // So we don't use a Lambda in the jsx.
  const changeLanguage = (language: string) => () =>
    handleChangeLanguage(language);

  return (
    <Dropdown
      className="link item right LanguageSwitcher"
      text={getKey('current.language', messages)}
    >
      <Dropdown.Menu>
        <Dropdown.Item
          title="FranÃ§ais"
          as={Button}
          onClick={changeLanguage('fr')}
        >
          FR
        </Dropdown.Item>
        <Dropdown.Item
          title="Deutsch"
          as={Button}
          onClick={changeLanguage('de')}
        >
          DE
        </Dropdown.Item>
        <Dropdown.Item
          title="Italianno"
          as={Button}
          onClick={changeLanguage('it')}
        >
          IT
        </Dropdown.Item>
        <Dropdown.Item
          title="English"
          as={Button}
          onClick={changeLanguage('en')}
        >
          EN
        </Dropdown.Item>
      </Dropdown.Menu>
    </Dropdown>
  );
};

export default LanguageSwitcher;
