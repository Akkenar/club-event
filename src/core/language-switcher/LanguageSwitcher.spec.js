import React from 'react';
import { render } from 'react-testing-library';
import LanguageSwitcher from './LanguageSwitcher';
import { BrowserRouter } from 'react-router-dom';
import LanguageContext from '../intl/LanguageContext';

const context = {
  messages: {},
  language: 'en',
  handleChangeLanguage: jest.fn(),
};

const LanguageSwitcherWithRouter = () => {
  return (
    <BrowserRouter>
      <LanguageContext.Provider value={context}>
        <LanguageSwitcher />
      </LanguageContext.Provider>
    </BrowserRouter>
  );
};

describe('LanguageSwitcher', () => {
  it('should match snapshot', () => {
    const wrapper = render(<LanguageSwitcherWithRouter />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should switch language when activating a link', () => {
    const wrapper = render(<LanguageSwitcherWithRouter />);

    const french = wrapper.queryByText('FR');
    french.click();
    expect(context.handleChangeLanguage).toHaveBeenCalledWith('fr');
  });
});
