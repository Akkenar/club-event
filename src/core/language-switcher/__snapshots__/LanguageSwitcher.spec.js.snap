// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher should match snapshot 1`] = `
<body>
  <div>
    <div
      aria-expanded="false"
      class="ui dropdown link item right LanguageSwitcher"
      role="listbox"
      tabindex="0"
    >
      <div
        aria-live="polite"
        class="text"
        role="alert"
      >
        current.language
      </div>
      <i
        aria-hidden="true"
        class="dropdown icon"
      />
      <div
        class="menu transition"
      >
        <a
          class="item"
          href="/fr"
          role="option"
          title="FranÃ§ais"
        >
          FR
        </a>
        <a
          class="item"
          href="/de"
          role="option"
          title="Deutsch"
        >
          DE
        </a>
        <a
          class="item"
          href="/it"
          role="option"
          title="Italianno"
        >
          IT
        </a>
        <a
          class="item"
          href="/en"
          role="option"
          title="English"
        >
          EN
        </a>
      </div>
    </div>
  </div>
</body>
`;
